pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
             cat(schemaString)
           })
 
+#' Get schema object
+#' 
+#' Returns the schema of this DataFrame as a list of named lists. Each named
```

Comment:
Ahhhh, it would essentially be a way to still get real values of certain types, but only when we want to as opposed to having the backend always deserialize it. So we could even just do some S3 functions that would handle the Spark types to get the info we need in different scenarios, right? Something that would end up looking like:
```R
schemaObj <- callJMethod(x@sdf, "schema")
out <- fields(schemaObj) # fields will call the appropriate schema methods and return only what we need
```
And we could have several different functions that mirror the methods of the Spark types that we need. Does that sound about right?